packages/presentation/lib/screen/home/home_bloc.dart

```diff
@@ -1,33 +1,72 @@
+import 'package:domain/enum/movie_type.dart';
+import 'package:domain/usecase/get_movie_response_usecase.dart';
 import 'package:presentation/base/bloc.dart';
+import 'package:presentation/screen/home/home_screen.dart';
+import 'package:presentation/screen/home/movie_model.dart';
+import '../details/details_page.dart';
 import 'home_data.dart';
+import 'home_view_mapper.dart';
 
-abstract class HomeBloc extends Bloc {
-  factory HomeBloc() => _HomeBloc();
+abstract class HomeBloc extends Bloc<HomeScreenArguments, HomeData> {
+  factory HomeBloc(
+    GetMovieResponseUseCase getMovieResponseUseCase,
+    HomeViewMapper viewMapper,
+  ) =>
+      _HomeBloc(
+        getMovieResponseUseCase,
+        viewMapper,
+      );
 
-  void onItemTapped(int index);
+  void openDetailsPage(MoviePresentation movie);
 }
 
-class _HomeBloc extends BlocImpl implements HomeBloc {
-  int? selectedIndex = 1;
-  final _screenData = HomeData.init();
+class _HomeBloc extends BlocImpl<HomeScreenArguments, HomeData>
+    implements HomeBloc {
+  HomeData _screenData = HomeData.init();
+  final HomeViewMapper _viewMapper;
 
-  _HomeBloc();
+  final GetMovieResponseUseCase _getMovieResponseUseCase;
+
+  _HomeBloc(
+    this._getMovieResponseUseCase,
+    this._viewMapper,
+  );
 
   @override
   void initState() {
     super.initState();
-    _updateData();
+    getInitialData();
+  }
+
+  void getInitialData() async {
+    _updateData(isLoading: true);
+    final responseTrending = await _getMovieResponseUseCase(MovieType.trending);
+    final responseAnticipated =
+        await _getMovieResponseUseCase(MovieType.anticipated);
```

Comment:
didn't notice before, u shouldn't load 2 lists while the page opens. u should load only one of them - list which will be displayed

Reply:
okay :(